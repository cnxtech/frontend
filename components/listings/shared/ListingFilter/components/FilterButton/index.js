import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import {
  StyledFilterButton,
  ChildrenWrapper
} from './styles'

class FilterButton extends PureComponent {
  render() {
    return (
      <StyledFilterButton
        mr={2}
        mb={2}
        px={[2, 3]}
        active={this.props.active}
        open={this.props.open}
        onClick={this.props.onClick}
      >
        {this.props.children}
      </StyledFilterButton>
    )
  }
}

FilterButton.propTypes = {
  active: PropTypes.bool.isRequired,
  onClick: PropTypes.func
}

export default FilterButton
