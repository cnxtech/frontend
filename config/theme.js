import theme from '@emcasa/ui'
import {fas} from '@fortawesome/pro-solid-svg-icons'
import {fal} from '@fortawesome/pro-light-svg-icons'
import icons from './icons'

export default {
  ...theme,
  icons: {emcasa: icons, default: fas, light: fal}
}
