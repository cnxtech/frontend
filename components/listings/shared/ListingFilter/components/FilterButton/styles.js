import styled from 'styled-components'
import theme from '@emcasa/ui'
import Button from '@emcasa/ui-dom/components/Button'
import { isMobile } from 'lib/mobile'

const StyledFilterButton = styled(Button)`
  position: relative;
  height: 32px;
  margin: 0 ${theme.space[2]}px ${theme.space[1]}px 0;
  border: 1px solid ${theme.colors.pink};
  background-color: ${({open}) => open ? theme.colors.pink : null};
  color: ${({active, open}) => active || open ? theme.colors.white : theme.colors.pink};
  font-size: ${theme.fontSizes[1]}px;

  ${({open}) => (open && !isMobile()) && `z-index: 1;`}

  &:last-child {
    margin-right: 0;
  }

  ${({open}) => open ? `
    &::before {
      content: '';
      position: absolute;
      z-index: 1;
      bottom: -26px;
      left: 50%;
      width: 15px;
      height: 15px;
      transform: translateX(-50%) rotate(-225deg);
      background-color: ${theme.colors.white};
      box-shadow: 0px 0px 8px -1px rgba(0,0,0,0.2);

      @media screen and (max-width: ${theme.breakpoints[0]}) {
        content: unset;
      }
    }
  `: null}

`

export {
  StyledFilterButton
}
