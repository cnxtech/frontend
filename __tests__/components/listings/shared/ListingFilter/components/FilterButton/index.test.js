import React from 'react'
import FilterButton from 'components/listings/shared/ListingFilter/components/FilterButton'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import theme from '@emcasa/ui'
import { ThemeProvider } from 'styled-components'

describe('<FilterButton/>', () => {
  it('should render component', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <FilterButton
          value={1}
          active={false}
          onClick={() => {}}
        />
      </ThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
