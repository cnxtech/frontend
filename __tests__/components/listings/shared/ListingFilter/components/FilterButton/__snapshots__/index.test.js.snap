// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterButton/> should render component 1`] = `
.c1 {
  outline: none;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #e0e6ed;
  background-color: #ffffff;
  border-radius: 4px;
  border-width: 1px;
  width: auto;
  height: 40px;
  margin-bottom: 10px;
  margin-right: 10px;
  padding-left: 10px;
  padding-right: 10px;
  color: #3c4858;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c1 :hover {
  background-color: #ffffff;
}

.c0 {
  position: relative;
  height: 32px;
  margin: 0 10px 4px 0;
  border: 1px solid #f50057;
  color: #f50057;
  font-size: 14px;
}

.c0:last-child {
  margin-right: 0;
}

@media screen and (min-width:780px) {
  .c1 {
    padding-left: 14px;
    padding-right: 14px;
  }
}

<button
  className="c0 c1"
  fontSize="default"
  height="medium"
  onClick={[Function]}
/>
`;
